mod/metrics/lib/javascript/script_metrics.js.coffee

```diff
@@ -19,23 +19,46 @@ decko.slotReady (slot) ->
   $('body').on 'change', ".TYPE-metric .card-editor.RIGHT-hybrid input[type=\'checkbox\']", (event) ->
     showResearchAttributes($(event.target))
 
+  $('td.metric-weight input').on 'keyup', (event) ->
+    activeEqualize()
+
+  $('#equalizer').on 'click', (event) -> 
+    if $(this).prop('checked') == true 
+      toEqualize( $('.wikiRating-editor') )
+
 showResearchAttributes = (checkbox) ->
   form = checkbox.closest("form")
+
   show_or_hide = checkbox.prop "checked"
   $.each ["value_type", "research_policy", "report_type", "methodology"], (_i, key) ->
     form.find(".card-editor.RIGHT-" + key).toggle show_or_hide
-
+    
 
 decko.editorContentFunctionMap['.pairs-editor'] = ->
   JSON.stringify pairsEditorHash(this)
 
 pairsEditorHash = (table) ->
   hash = {}
-  table.find("tbody tr").each ->
+  variableMetricRows(table).each ->
     cols = $(this).find('td')
     if (key = $(cols[0]).data('key'))
       hash[key] = $(cols[1]).find('input').val()
-  hash
+  hash 
+
+# if all values are equals return "true"
+valueFormEqual = (table) -> 
+  console.log('table', table)
+  aux = false 
+  values = []
```

Comment:
I suggest a separate method for this.  Eg.  "allVariableValues"

If you make one function `allVariableValues` that returns values, and a second function `allValuesAreEqual`, then you should be able to write a unit test for that second function, because it won't need the DOM.